eras/shelley/impl/src/Cardano/Ledger/Shelley/AdaPots.hs

```diff
@@ -72,6 +93,87 @@ totalAdaES cs =
         reservesAdaPot,
         rewardsAdaPot,
         utxoAdaPot,
+        -- keyDepositAdaPot,  -- We don't count these two, as their
+        -- poolDepositAdaPot, -- sum is always depositsAdaPot
         depositsAdaPot,
         feesAdaPot
       } = totalAdaPotsES cs
+
+-- =============================================
+-- Produced and Consumed are specialized AdaPots
+-- relative to the actions of a TxBody
+
+-- | Itemizing what is consumed by a transaction
+data Consumed = Consumed
+  {conInputs :: !Coin, conRefunds :: !Coin, conWithdrawals :: !Coin}
+
+instance Show Consumed where
+  show (Consumed (Coin i) (Coin r) (Coin w)) =
+    "Consumed(Inputs " ++ show i ++ ", Refunds " ++ show r ++ ", Withdrawals " ++ show w ++ ") = " ++ show (i + r + w)
+
+-- | Itemizing what is Produced by a transaction
+data Produced = Produced
+  {proOutputs :: !Coin, proFees :: Coin, proDeposits :: !Coin}
```

Comment:
Was `proFees` left lazy on purpose. If so, we should have a comment on why this is necessary, otherwise someone will "fix" it eventually. On the other hand if it doesn't have to be lazy then it is better to make it strict

Reply:
you are right.